### PR TITLE
PBI #650 - Sarine Profile – Stone Properties – Order, Appearance and Format 

### PR DESCRIPTION
**Pull request by vlad-snrg**
https://sarinevso.visualstudio.com/DefaultCollection/SarineProfile/_workitems?_a=edit&id=650

Html: 
- added data attributes to html markup : 
 1. `"data-properties"` - to stone properties container
 2. `"data-property"` - to each stone properties

JS:
- added js handlers: 
 1. for formatting stones properties values
 2. for enable stone properties in markup 

For enabling stone properties in specific order: 
 1. add "stoneProperties" array to your configurator with objects, with following object template:
  1. `"key"` - name of property, which equal "data-property" attribute in config/template.html
  2. `"order"` - property order in markup
  3. ~~`"row"` - also you can specify the row. Available values: `0`, `1`, `2`~~
  4. `"format"` - numeric value, which specify number of decimals

Example: 
```json
"stoneProperties": [
 {"key": "sarineId", "order": 1},
 {"key": "gradedBy", "order": 2, "row":2},
 {"key": "carat", "order": 2, "row":3, "format":3},
]
```
You can see all changes here: https://api-5.qa.sarine.com/viewer/v1/139D33GQHY/PBI650C2

### CODE REVIEW

**Comment by vlad-snrg** [comment]
Changed stone properties handler: 
 - remove `row` property from configurator
 - add stone properties 4 per row

File: app/js/app.responsive.js [hunk 1]
```diff
@@ -111,13 +111,25 @@
...
             isWidgetLoaded = false,
             isStoneRound = stone && stone.stoneProperties && (stone.stoneProperties.shape === 'Round' || stone.stoneProperties.shape === 'ModifiedRound'),
             hasSummaryStoryline = false,
             summaryStorylineClass = 'summary-storyline',
             disableTabIndexTags = ['canvas', 'button', 'svg', 'a', 'img'],
             isReportWidget = wConfig.report,
-            parentHostname = (window.location !== window.parent.location) ? tplUtils.getParameter('hostname') : 'direct';
+            parentHostname = (window.location !== window.parent.location) ? tplUtils.getParameter('hostname') : 'direct',
+            defaultStonePropertiesConfig = [
+                {key: "sarineId", order: 1, row: 0},
```
Comment (adica) [inline app/js/app.responsive.js]: @vlad-snrg 
why you kept the "row"? i though you removed it as described.

File: app/js/app.responsive.js [hunk 2]
```diff
@@ -1084,6 +1145,64 @@
...
+            function enableStoneProperties() {
+                var spConfig = stonePropertiesConfig,
+                    propertyAttr = {
+                        unit: 'data-property',
+                        container: 'data-properties'
+                    },
+                    $tmpContainer = $('<div/>'),
+                    $propertyContainers = $('[' + propertyAttr.container + ']');
+
+                var MAX_ELEMENTS_PER_ROW = 4;
```
Comment (adica) [inline app/js/app.responsive.js]: @vlad-snrg 
what about mobile and tablets? how to you change it?

File: app/js/app.responsive.js (hunk 1 above)
Comment (a-klyuchnik) [inline app/js/app.responsive.js]: @adica 
We removed it from the user accessible configuration. It's a good tool for us to use it for rows handling.

File: app/js/app.responsive.js (hunk 2 above)
Comment (vlad-snrg) [inline app/js/app.responsive.js]: this changes work with mobile and tablets too

Comment (adica) [inline app/js/app.responsive.js]: @vlad-snrg 
so this "MAX_ELEMENTS_PER_ROW" always equals to 4?
do you decrease it on mobile or not?

Comment (vlad-snrg) [inline app/js/app.responsive.js]: we don`t decrease it

**Comment by vlad-snrg** [comment]
Changes: 
 - show `graded by` property accordingly PBI#527
 - always show `information provided by` property

**Pull request merged**